File: src/main/java/org/parthvnp/Array/UglyNumber.java
package org.parthvnp.Array;

public class UglyNumber {
  public boolean isUgly(int n) {
    if (n <= 0) return false;
    int[] x = new int[] {2, 3, 5};
    for (int i : x) n = kd(n, i);
    return n == 1;
  }

  public int kd(int dividend, int divisor) {
    while (dividend % divisor == 0) dividend = dividend / divisor;
    return dividend;
  }

  public static void main(String[] args) {
    var s = new UglyNumber();
    System.out.println(s.isUgly(6));
    System.out.println(s.isUgly(8));
    System.out.println(s.isUgly(14));
  }
}

File: src/main/python/arrays/ugly_number.py
from shared.test import test


class IsUgly:
    def isUgly(self, n: int) -> bool:
        def keepDividingWhenDivisible(dividend, divisor):
            while dividend % divisor == 0: dividend = dividend / divisor
            return dividend

        for i in (2, 3, 5):
            n = keepDividingWhenDivisible(n, i)
        return n == 1


if __name__ == '__main__':
    s = IsUgly()
    e1 = s.isUgly(6)
    test(True, e1)
    e2 = s.isUgly(8)
    test(True, e2)
    e3 = s.isUgly(14)
    test(False, e3)
